### PR TITLE
Make anchor behavior more intuitive

### PR DESCRIPTION
**Pull request by fannieyan**

### CODE REVIEW

File: Extensions/AnchorBehavior/anchorruntimebehavior.ts [hunk 1]
```diff
@@ -15,14 +15,18 @@ namespace gdjs {
...
       super(runtimeScene, behaviorData, owner);
-      this._relativeToOriginalWindowSize = !!behaviorData.relativeToOriginalWindowSize;
+      this._relativeToOriginalWindowSize =
+        !!behaviorData.relativeToOriginalWindowSize;
       this._leftEdgeAnchor = behaviorData.leftEdgeAnchor;
       this._rightEdgeAnchor = behaviorData.rightEdgeAnchor;
       this._topEdgeAnchor = behaviorData.topEdgeAnchor;
       this._bottomEdgeAnchor = behaviorData.bottomEdgeAnchor;
+      this._useLegacyBottomAndRightAnchors =
+        behaviorData.useLegacyWindowAnchors === undefined ? true : false;
```
Comment (4ian) [inline Extensions/AnchorBehavior/anchorruntimebehavior.ts]:
```suggestion
        behaviorData.useLegacyWindowAnchors === undefined ? true : behaviorData.useLegacyWindowAnchors;
```

File: Extensions/AnchorBehavior/anchorruntimebehavior.ts [hunk 2]
```diff
@@ -262,28 +273,91 @@ namespace gdjs {
...
+              );
+            }
+            if (
+              this._rightEdgeAnchor !==
+              AnchorRuntimeBehavior.HorizontalAnchor.NONE
+            ) {
+              this.owner.setX(
+                bottomRightCoord[0] +
+                  this.owner.getX() -
+                  this.owner.getDrawableX()
```
[the remark this answers is not in the document]
Comment (fannieyan) [reply Extensions/AnchorBehavior/anchorruntimebehavior.ts]: I opened the PR a little early 😅 But yes, the width needs to be substracted here

File: Extensions/AnchorBehavior/anchorruntimebehavior.ts [hunk 3]
```diff
@@ -262,28 +273,91 @@ namespace gdjs {
...
-          );
+        // End of compatibility code
+        else {
+          // Resize if right and left anchors are set
+          if (
+            this._rightEdgeAnchor !==
+              AnchorRuntimeBehavior.HorizontalAnchor.NONE &&
+            this._leftEdgeAnchor !== AnchorRuntimeBehavior.HorizontalAnchor.NONE
+          ) {
+            this.owner.setWidth(bottomRightCoord[0] - topLeftCoord[0]);
```
[the remark this answers is not in the document]
Comment (fannieyan) [reply Extensions/AnchorBehavior/anchorruntimebehavior.ts:314]: Actually, I do

**Review by 4ian — APPROVED** [review]

File: Extensions/AnchorBehavior/tests/anchorruntimebehavior.spec.js [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+      runtimeGame.setGameResolutionSize(2000, 2000);
+      runtimeScene.renderAndStep(1000 / 60);
+
+      expect(object.getX()).to.equal(500);
+      expect(object.getY()).to.equal(500);
+      expect(object.getHeight()).to.equal(1010);
+    });
+
+    it('anchors the bottom edge of object (proportional)', function () {
+      var object = createObject();
```
Comment (4ian) [inline Extensions/AnchorBehavior/tests/anchorruntimebehavior.spec.js:205]: Uber nitpicking: `var` => `const`

File: Extensions/AnchorBehavior/anchorruntimebehavior.ts [hunk 5]
```diff
@@ -368,6 +368,24 @@ namespace gdjs {
 
     doStepPostEvents(runtimeScene) {}
 
+    // Set anchors for testing purposes.
+    setAnchor(edge, anchorType: number) {
```
Comment (4ian) [inline Extensions/AnchorBehavior/anchorruntimebehavior.ts:372]: As discussed let's remove it in favor of a parameter in createObject

File: Extensions/AnchorBehavior/tests/anchorruntimebehavior.spec.js [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+      runtimeGame.setGameResolutionSize(1000, 1000);
+      object.setPosition(500, 500);
+
+      object.getBehavior('Anchor').setAnchor('top', 3);
```
Comment (4ian) [inline Extensions/AnchorBehavior/tests/anchorruntimebehavior.spec.js]: Same question between the name of the test and the actual call to setAnchor('top')?

**Pull request merged**